alacritty_terminal/src/term/mod.rs

```diff
@@ -938,6 +888,58 @@ impl<T> Term<T> {
         res
     }
 
+    /// Check selection state of a cell.
+    pub fn is_selected(
```

Comment:
since it's on `Term` now maybe we can use it somewhere else, but that cursor stuff annoys me a bit. Passing selection also sounds questionable, but I think it's fine, if we hide this function for now...

Reply:
Why move it on `Term`, it seems out of place. Passing `selection` to something on `Term` doesn't make a ton of sense, because it already has selection stored on it. So that makes all of this rather questionable.

Also passing a cursor to a function that checks if a cell is selected doesn't really make sense to me for a terminal selection check?

Reply:
Because I can't create a renderable cell otherwise? I need selection and searcher to perform that. I'd like to not do so, but then I need to change the logic entirely, though, maybe selection logic doesn't really matter for is_selected. The main issue is that it originates from renderable cell constructor, and i need it to create a cursor...